drivers/isl29125/isl29125.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2015 Ludwig Ortmann
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_isl29125
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the ISL29125 RGB light sensor
+ *
+ * @author      Ludwig Ortmann <ludwig.ortmann@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "isl29125.h"
+#include "isl29125-internal.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/***********************************************************************
+ * internal API declaration
+ **********************************************************************/
+
+static inline char parse_mode(isl29125_mode_t mode);
+static inline char parse_range(isl29125_range_t range);
+static inline char parse_resolution(isl29125_resolution_t resolution);
+static inline char parse_conf1(isl29125_mode_t mode, isl29125_range_t range,
+        isl29125_resolution_t resolution);
+
+/***********************************************************************
+ * internal API implementation
+ **********************************************************************/
+
+static inline char parse_mode(isl29125_mode_t mode)
```

Comment:
Ah, I remember, I decided against this because it would require including the internal into the public header.